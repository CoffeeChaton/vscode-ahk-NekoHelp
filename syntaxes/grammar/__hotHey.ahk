#Warn All, MsgBox

(:: foo("case (")
~[:: foo("case [")
;^Avoid vscode automatic bracket matching, here if it is Str

~F12::Reload

foo(case) {
    MsgBox, % "i am foo, and use at " . case
}


F1::MsgBox, % A_ThisHotkey

w & s::
    MsgBox, % A_ThisHotkey
return

a Up::
    MsgBox, % A_ThisHotkey
return

c & b::
    MsgBox, % A_ThisHotkey
return

F10 & F11 Up::
    MsgBox, % A_ThisHotkey
return

