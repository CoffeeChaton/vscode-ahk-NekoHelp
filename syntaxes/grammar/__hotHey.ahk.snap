>; normal
#^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>(:: foo("case (")
#^ source.ahk meta.hotkey.line.ahk string.2.storage.modifier.hotkey.ahk
# ^^ source.ahk meta.hotkey.line.ahk
#   ^ source.ahk
#    ^^^ source.ahk entity.name.function.ahk
#       ^ source.ahk punctuation.bracket.ahk
#        ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#         ^^^^^^ source.ahk string.quoted.double.ahk
#               ^ source.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                ^ source.ahk punctuation.bracket.ahk
>~[:: foo("case [")
#^ source.ahk meta.hotkey.line.ahk storage.modifier.hotkey.ahk
# ^ source.ahk meta.hotkey.line.ahk string.2.storage.modifier.hotkey.ahk
#  ^^ source.ahk meta.hotkey.line.ahk
#    ^ source.ahk
#     ^^^ source.ahk entity.name.function.ahk
#        ^ source.ahk punctuation.bracket.ahk
#         ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#          ^^^^^^ source.ahk string.quoted.double.ahk
#                ^ source.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                 ^ source.ahk punctuation.bracket.ahk
>;^Avoid vscode automatic bracket matching, here if it is Str
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>~F12::Reload
#^ source.ahk meta.hotkey.line.ahk storage.modifier.hotkey.ahk
# ^^^ source.ahk meta.hotkey.line.ahk string.2.storage.modifier.hotkey.ahk markup.bold.hotkey.ahk
#    ^^ source.ahk meta.hotkey.line.ahk
#      ^^^^^^ source.ahk keyword.control.flow.ahk
>
>foo(case) {
#^^^ source.ahk entity.name.function.ahk
#   ^ source.ahk punctuation.bracket.ahk
#    ^^^^ source.ahk keyword.control.flow.ahk
#        ^ source.ahk punctuation.bracket.ahk
#         ^ source.ahk
#          ^ source.ahk punctuation.bracket.ahk
>    MsgBox, % "i am foo, and use at " . case
#^^^^ source.ahk
#    ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#          ^ source.ahk meta.command.line.ahk punctuation.ahk
#           ^ source.ahk meta.command.line.ahk
#            ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#             ^ source.ahk meta.command.line.ahk
#              ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#               ^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                                    ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                                     ^^^^^^^ source.ahk meta.command.line.ahk
>}
#^ source.ahk punctuation.bracket.ahk
>